src/OpenTelemetry.Exporter.Geneva/MsgPackExporter/MsgPackLogExporter.cs

```diff
@@ -202,9 +204,22 @@ internal int SerializeLogRecord(LogRecord logRecord)
         }
 
         // Part A - core envelope
-        cursor = AddPartAField(buffer, cursor, Schema.V40.PartA.Name, eventName);
 
-        cntFields += 1;
+        var eventId = logRecord.EventId;
+        bool hasEventId = eventId != default;
+
+        if (hasEventId && (this.m_eventNameExportMode & EventNameExportMode.ExportAsField) != 0 && !string.IsNullOrWhiteSpace(eventId.Name))
```

Comment:
(this.m_eventNameExportMode & EventNameExportMode.ExportAsField) != 0

or

this.m_eventNameExportMode == EventNameExportMode.ExportAsField

Reply:
I was trying to make it future-proof. We might extend the enum to add an option to export `eventId.Name` as the table name. Technically, one could select both the options when that happened, so I didn't want to do a simple equality check.

Reply:
I see (and also noticed the `[Flags]` part), thanks for confirmation.